src/main/java/configurationslicing/promotedbuilds/PromotedBuildsNameSlicer.java

```diff
@@ -96,26 +99,26 @@ public boolean setValues(AbstractProject<?,?> item, List<String> rawValues) {
                 }
             }
 
-            if(rawValues.get(0).equals(NOTHING) && property != null) {
-                property.getActiveItems().removeAll(property.getActiveItems());
+            if(rawValues.get(0).equals(NOTHING)) {
+                property.getItems().removeAll(property.getActiveItems());
                 return true;
             }
 
-            Map<String, PromotionProcess> oldPromotions = new HashMap<String, PromotionProcess>();
-            for(PromotionProcess promotionProcess : property.getActiveItems()) {
+            Map<String, PromotionProcess> oldPromotions = Maps.newHashMap();
+            for(PromotionProcess promotionProcess : property.getItems()) {
                 oldPromotions.put(promotionProcess.getName(), promotionProcess);
             }
 
             // Remove old promotions
```

Comment:
large method, consider extracting this and the next block of code into their own methods with suitable names instead of comments to explain what they do